src/operator/databaseconfigurator/db_helpers.go

```diff
@@ -0,0 +1,23 @@
+package databaseconfigurator
+
+import (
+	"fmt"
+	"strings"
+)
+
+const PGUsernameMaxLength = 63
+
+func BuildPostgresUsername(clusterUID string, workloadName, namespace string) string {
+	username := fmt.Sprintf("%s_%s_%s", KubernetesToPostgresName(workloadName), KubernetesToPostgresName(namespace), KubernetesToPostgresName(clusterUID))
+	if len(username) > PGUsernameMaxLength {
+		username = username[:PGUsernameMaxLength]
```

Comment:
You might be trimming the clusterUID here, which means that your name is going to suffer from uniqueness issues. 
This didn't use to happen when this code ran in Otterize Cloud, because the first chunk of the name was the otterize service ID, which guaranteed uniqueness on its own. This is not the case anymore, and so you should make sure you add a unique id, or suffix the trimmed name with a hash over the full username (we use a similar method in the IAM policy agents, look for `TruncateHashName`). 

Also, I can already spot a potential issue with Postgres visibility here. We use the postgres username in postgres visibility to detect to which service this traffic belongs. We need to make sure we somehow preserve this functionality even when the usernames are generated by the credentials operator.